daemon/dns/dns_resolvconf_file.go

```diff
@@ -94,3 +101,85 @@ func restoreDNS() error {
 	}
 	return internal.FileWrite(resolvconfFilePath, out, internal.PermUserRWGroupROthersR)
 }
+
+func tryToRestoreDNS() {
+	// if target is symlink, probably it is managed by other software - do nothing
+	if internal.IsSymLink(resolvconfFilePath) {
+		return
+	}
+	// if target /etc/resolv.conf contains nordvpn changes:
+	// if possible restore from backup
+	// if backup does not exists, create simple dns settings file
+	out, err := internal.FileRead(resolvconfFilePath)
+	if err != nil {
+		log.Println(internal.ErrorPrefix, fmt.Errorf("reading resolv.conf: %w", err))
+		return
+	}
+	if !strings.Contains(string(out), resolvconfFileMark) {
+		return
+	}
+
+	log.Println(internal.WarningPrefix, "/etc/resolv.conf contains our changes - need to fix this")
+
+	// try to unlock, if file was locked
+	_ = internal.FileUnlock(resolvconfFilePath)
+	// restore from backup if backup file exists
+	if internal.FileExists(resolvconfBackupPath) {
+		log.Println(internal.WarningPrefix, "resolv.conf backup found - restore from it")
+		backup, err := internal.FileRead(resolvconfBackupPath)
+		if err != nil {
+			log.Println(internal.ErrorPrefix, fmt.Errorf("reading backup resolv.conf: %w", err))
+		} else {
+			if err := internal.FileWrite(resolvconfFilePath, backup, internal.PermUserRWGroupROthersR); err != nil {
+				log.Println(internal.ErrorPrefix, fmt.Errorf("restore from backup resolv.conf: %w", err))
+			}
+			return
+		}
+		// remove backup
+		internal.FileDelete(resolvconfBackupPath)
```

Comment:
If the backup file exists maybe should always delete it?

Reply:
yes, right, this was my mistake/leftover.